File: Subless.Models/SubscriptionStatus.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Subless.Models
{
    public class SubscriptionStatus
    {
        public bool IsActive { get; set; }
        public bool IsCancelled { get; set; }
        public DateTimeOffset? BillingDate {get;set;}
        public DateTimeOffset? CancellationDate { get; set; }
        public DateTimeOffset? SubscriptionEndDate { get; set; }
    }
}

File: Subless.UI/sublessui/src/app/models/SubscriptionStatus.ts
import { ISubscriptionStatus } from "./ISubscriptionStatus";

export class SubscriptionStatus implements ISubscriptionStatus {
    constructor(
        public isActive: boolean,
        public isCancelled: boolean,
        public billingDate: Date,
        public cancellationDate: Date,
        public subscriptionEndDate: Date,
    ) { }

}